xbmc/filesystem/ZipFile.cpp

```diff
@@ -305,7 +305,7 @@ ssize_t CZipFile::Read(void* lpBuf, size_t uiBufSize)
   {
     uLong iDecompressed = 0;
     uLong prevOut = m_ZStream.total_out;
-    while (((int)iDecompressed < uiBufSize) && ((m_iZipFilePos < mZipItem.csize) || (m_bFlush)))
+    while ((iDecompressed < uiBufSize) && ((m_iZipFilePos < mZipItem.csize) || (m_bFlush)))
```

Comment:
doesn't this need a cast as well? uLong is uint32_t and uiBufSize is size_t which can be uint64_t
same goes for the second case